chatmemory implementation

enricorampazzo opened the pull request:
This is the same as [this PR](https://github.com/spring-projects/spring-ai/pull/2063), but with a signed-off commit

ilayaperumalg commented:
@meistermeier Could you help reviewing this PR?

meistermeier commented on spring-ai-docs/src/main/antora/modules/ROOT/pages/api/chatclient.adoc:
```diff
@@ -374,7 +374,7 @@ Refer to the xref:_retrieval_augmented_generation[Retrieval Augmented Generation
-There are currently two implementations, `InMemoryChatMemory` and `CassandraChatMemory`, that provide storage for chat conversation history, in-memory and persisted with `time-to-live`, correspondingly.
+There are currently three implementations, `InMemoryChatMemory`, `CassandraChatMemory` and `Neo4jChatMemory`, that provide storage for chat conversation history, in-memory, persisted with `time-to-live` in Cassandra, and persisted without `time-to-live` in Neo4j correspondingly.
 
 To create a `CassandraChatMemory` with `time-to-live`:
```
Maybe a more general remark, but the sections for Cassandra and Neo4j as chat memory stores(?) should look the the same regarding parameters etc. (maybe this is also true for `InMemoryChatMemory` and this needs to be added).

enricorampazzo replied:
you mean that I should implement TTL?

meistermeier replied:
No, it was just about having the same (or comparable table) for Cassandra and Neo4j. Because now Neo4j has this property table but Cassandra not. Only a documentation thing.

meistermeier reviewed (changes requested):
I have no deep insights how a chat memory should work and what the expected outcome is, besides from what I can read up in code here and compared to e.g. Cassandra store. Because of this I would be happy to get also your thoughts on the overall behaviour @ilayaperumalg (and team).
Some things I am missing here:
* a dedicated integration test within the store module for the chat memory
* spring formatter (build fails)

enricorampazzo commented on vector-stores/spring-ai-neo4j-store/src/main/java/org/springframework/ai/chat/memory/neo4j/Neo4jChatMemoryConfig.java:
```diff
@@ -0,0 +1,192 @@
+
+	private final String mediaLabel;
+
+	public String getSessionLabel() {
```
If I remove these getters, how can I access these values [here on line 47?](https://github.com/spring-projects/spring-ai/pull/2071/commits/10089b13a4ed6c812da80f15fe0d99bcb482bd27#diff-0ce2b6139ff75b362655108505f6589cb91bbb2f19eb6ef9dbfb96e929bf89d0R56)

meistermeier replied:
Please don't ask how I missed this. Sorry for the noise.

ilayaperumalg commented:
@enricorampazzo Thanks for the PR contributing chat memory implementation! @meistermeier Thanks for the review!

The changes look good. Rebased the PR, refactored the auto configurations based on the latest auto-configuration updates, fixed some cosmetic issues, merged as 1d4bde80e59be5e36ee6070b9b6b5c9f98a93197

enricorampazzo commented:
> @enricorampazzo Thanks for the PR contributing chat memory implementation! @meistermeier Thanks for the review!
> 
> The changes look good. Rebased the PR, refactored the auto configurations based on the latest auto-configuration updates, fixed some cosmetic issues, merged as [1d4bde8](https://github.com/spring-projects/spring-ai/commit/1d4bde80e59be5e36ee6070b9b6b5c9f98a93197)

Thank you. Sorry to bother you, but would it be possible to change my git username in the commit you merged to "enricorampazzo"? unfortunately it was wrong on my end, and I would like to be mentioned as a Spring AI contributor, if it is not too much work on your end, thanks :)

ilayaperumalg commented:
@enricorampazzo I am so sorry that the GH username was incorrect. I wish I could help here by amending the GH commit but unfortunately, we can not force push the changes into "main" as it will impact other forks go out of sync. Thank you for your contributions. I had also ensured your name was updated as author in the files related to the PR when I originally merged the PR and I will make sure to mention your username when we publish the M7 release as the list of Spring AI contributors.

enricorampazzo commented:
@ilayaperumalg Ok, thanks :)
